app/controllers/patrons_controller.rb

```diff
@@ -5,22 +5,60 @@ def index
     @patrons = Patron.all
   end
 
+  def show
+    @patron = Patron.find(params[:id])
+  end
+
   def new
     @patron = Patron.new
   end
 
   def create
     @patron = Patron.new(patron_params)
-    if @patron.save
-      redirect_to patrons_path, notice: 'Patron added successfully.'
+    
+    respond_to do |format|
+      if @patron.save
+        format.html { redirect_to @patron, notice: 'Patron was successfully created.' }
+        format.json { render :show, status: :created, location: @patron }
+      else
+        format.html { render :new }
+        format.json { render json: @patron.errors, status: :unprocessable_entity }
```

Comment:
JSON is more like a universal translator for web data. we need to return data in JASON because It's a way of organizing data that makes it easy for different parts of a website to understand and work with each other.